src/components/parser/log.ts

```diff
@@ -319,11 +327,35 @@ export class Parser {
         this.showLinterDiagnostics(linterLog)
     }
 
-    showCompilerDiagnostics() {
+    async showCompilerDiagnostics() {
         this.compilerDiagnostics.clear()
         const diagsCollection: { [key: string]: vscode.Diagnostic[] } = {}
         for (const item of this.buildLog) {
-            const range = new vscode.Range(new vscode.Position(item.line - 1, 0), new vscode.Position(item.line - 1, 65535))
+            let startChar = 0
+            let endChar = 65535
+            if (item.errorPosText) {
+                // try to find the errorPosText in the respective line of the document
+                try {
+                    const document = await vscode.workspace.openTextDocument(item.file)
+                    const line = document.lineAt(item.line - 1)
+                    if (line) {
+                        let pos = line.text.indexOf(item.errorPosText)
+                        if (pos >= 0) {
+                            pos += item.errorPosText.length
+                            // find the length of the last word in the error
+                            // This is the length of the error-range
+                            const len = item.errorPosText.replace(/^.* (.*)/, '$1').length
+                            if (len > 0) {
+                                startChar = pos - len
```

Comment:
Doesn't affect performance too much, but `startChar = item.errorPosText.lastIndexOf(' ')` would be better than this regex that has to do a lot of backtracking.